spock-specs/src/test/groovy/org/spockframework/docs/datadriven/v7/MathSpec.groovy

```diff
@@ -0,0 +1,24 @@
+package org.spockframework.docs.datadriven.v7
+
+import spock.lang.Specification
+
+// tag::example[]
+class MathSpec extends Specification {
+  def "maximum of two numbers"() {
+    expect:
+    Math.max(a, b) == c
+    Math.max(d, e) == f
+
+    where:
+    a | b || c
+    1 | 3 || 3
+    7 | 4 || 7
+    0 | 0 || 0
+
+    ; d ; e ;; f ;
```

Comment:
why are there `;` before and after the row

Reply:
Just to showcase that you can.
This way you can visually add a left and right border so that it looks more table-like.
If my data table separator PR is merged you can even add an upper border and with a bit distance bottom border like
```groovy
    ______________
    ; d ; e ;; f ;
    ; 1 ; 3 ;; 3 ;
    ; 7 ; 4 ;; 7 ;
    ; 0 ; 0 ;; 0 ;
    ______________
```

Reply:
I think it is better to introduce one concept at a time and would move that into a separate example. In this case I think it makes more sense to have the same layout as the pipe table above, not every new user will directly see what is necessary and what is not.

Documentation is also treated as a kind of a style guide, and while it is possible to encase your table that way and we may mention it, I would not recommend it. Groovy is often referred to as java without semicolons, so having more visual noise is often frowned upon.

Reply:
Well, the example right before shows it without the borders:
![grafik](https://user-images.githubusercontent.com/325196/77592001-7c4fce00-6ef1-11ea-9f85-d1c67bca106e.png)
But I can of course add this as an additional example.

Reply:
Another reason why I combined them was, because the first semicolon in the first line must have a preceding statement, because you cannot end the label with a semicolon.
This is **not** valid Groovy syntax:
```groovy
    where:
    ; a ; b ;; c ;
    ; 1 ; 3 ;; 3 ;
    ; 7 ; 4 ;; 7 ;
    ; 0 ; 0 ;; 0 ;
```
With the other table in front it was valid.
So don't wonder that the new example will be
```groovy
    where:
    ____________=_
    ; a ; b ;; c ;
    ; 1 ; 3 ;; 3 ;
    ; 7 ; 4 ;; 7 ;
    ; 0 ; 0 ;; 0 ;
```
If the data table separator PR gets merged, the `=` can be replaced by an `_`, then it looks properly like
```groovy
    where:
    ______________
    ; a ; b ;; c ;
    ; 1 ; 3 ;; 3 ;
    ; 7 ; 4 ;; 7 ;
    ; 0 ; 0 ;; 0 ;
```

Reply:
I get more and more convinced to **not** include those examples at all, if they have all these edge cases. You can still use them in your projects if you find them helpful, but I don't think it is good to add them to the official documentation. We should present them to be a drop in replacement to the `|`/`||` style nothing more nothing less.

Reply:
"All these edge cases"?
Maybe I'm too tunnel-focused right now, but I see only one edge case, that the `where:` label must not be followed by a semicolon directly.
As the data table serparator PR was already merged, I now also adapted the example here to be the proper one without equals sign.
I rebased this PR and fixed the conflicts.
Maybe you want to have a look again now and decide then.
If you really want the example removed, I'll remove it, I just don't get what you mean with "All these edge cases".

Reply:
Well the other is that you can't do the same with `|`. I prefer to remove them from the documentation.

Reply:
You can:
```groovy
    where:
    ______________
    ; a | b || c ;
    ; 1 | 3 || 3 ;
    ; 7 | 4 || 7 ;
    ; 0 | 0 || 0 ;
```
(I know this is not what you meant :-) )

Reply:
done